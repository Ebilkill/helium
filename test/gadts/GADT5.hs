data X a where
    I :: Int -> X Int
    B :: Bool -> X Bool
    XEQ :: a -> a -> X Bool

f :: X a -> Bool
f (I i) = i == 3
f (B b) = b
f (XEQ x y) = x == y