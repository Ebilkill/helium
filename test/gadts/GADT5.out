Prelude is up to date
Compiling gadts/GADT5.hs
fromList []
Could not discharge [$Eq _32]


(7,4): Type error in pattern of function binding
 pattern          : I i
   type           : X Int 
   does not match : X Bool

Compilation failed with 1 error
